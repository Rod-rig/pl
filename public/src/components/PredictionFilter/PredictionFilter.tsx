import {
  createStyles,
  FormControl,
  InputLabel,
  MenuItem,
  Select,
  Theme,
  WithStyles,
  withStyles,
} from "@material-ui/core";
import { observer } from "mobx-react";
import * as React from "react";
import { IPredictionStore } from "../../@types";
import { constants } from "../../constants";
import { dict } from "../../dict";

const styles = ({ spacing }: Theme) =>
  createStyles({
    control: {
      margin: spacing.unit,
      width: 200,
    },
    wrap: {
      alignItems: "center",
      display: "flex",
    },
  });

interface IDatePickerProps extends WithStyles<typeof styles> {
  store: IPredictionStore;
}

export const PredictionFilter = withStyles(styles)(
  observer(
    class extends React.Component<IDatePickerProps, {}> {
      public render() {
        const { classes, store } = this.props;

        return (
          <div>
            <div className={classes.wrap}>
              <FormControl className={classes.control}>
                <InputLabel htmlFor="date">{dict.date_label}</InputLabel>
                <Select
                  value={store.currentDate}
                  onChange={this.handleDateChange}
                  inputProps={{
                    id: "date",
                    name: "date",
                  }}
                >
                  {store.dates.map((item: string) => {
                    const date = item
                      .split("-")
                      .reverse()
                      .join(".");
                    return (
                      <MenuItem key={item} value={item}>
                        {date}
                      </MenuItem>
                    );
                  })}
                </Select>
              </FormControl>

              <FormControl
                className={classes.control}
                disabled={store.matches.length < 1}
              >
                <InputLabel htmlFor="tournament">
                  {dict.tournament_label}
                </InputLabel>
                <Select
                  value={store.tournamentId}
                  onChange={this.handleTournamentChange}
                >
                  <MenuItem
                    key={constants.defaultTournamentsValue}
                    value={constants.defaultTournamentsValue}
                  >
                    All
                  </MenuItem>
                  {Object.keys(store.tournaments).map((id: string) => {
                    return (
                      <MenuItem key={id} value={id}>
                        {store.tournaments[id]}
                      </MenuItem>
                    );
                  })}
                </Select>
              </FormControl>
            </div>
          </div>
        );
      }

      private handleDateChange = (event: any) => {
        const { store } = this.props;
        const { cache } = store;

        store.setCurrentDate(event.target.value);

        store.setTournamentId(constants.defaultTournamentsValue);

        store.currentDate in cache
          ? store.setMatches(cache[store.currentDate])
          : store.fetchMatches();
      };

      private handleTournamentChange = (event: any) => {
        const { store } = this.props;
        const { cache, currentDate } = store;
        const cacheMatches = cache[currentDate];

        store.setTournamentId(event.target.value);

        const filteredMatches =
          store.tournamentId === constants.defaultTournamentsValue
            ? cacheMatches
            : store.filterMatches(cacheMatches);
        store.setMatches(filteredMatches);
      };
    },
  ),
);
