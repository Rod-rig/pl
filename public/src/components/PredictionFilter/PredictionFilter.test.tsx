import * as React from "react";
import * as renderer from "react-test-renderer";
import { PredictionFilter } from "./PredictionFilter";

describe("PredictionFilter", () => {
  const predictionFilter = (
    <PredictionFilter
      store={{
        cache: {},
        closeSuccessMsg: jest.fn(),
        currentDate: "date",
        dates: ["Berlin", "Rome", "Paris", "London"],
        fetchMatches: jest.fn(),
        fetchMatchesError: jest.fn(),
        fetchMatchesSuccess: jest.fn(),
        handleChange: jest.fn(),
        handleDateChange: jest.fn(),
        handleSubmit: jest.fn(),
        handleSubmitError: jest.fn(),
        handleSubmitSuccess: jest.fn(),
        handleTournamentChange: jest.fn(),
        isFetched: true,
        isLoaded: true,
        isSuccessSubmit: false,
        matches: [],
        setCurrentDate: jest.fn(),
        setMatches: jest.fn(),
        setTournamentId: jest.fn(),
        tournamentId: "",
        tournaments: {
          "0": "EPL",
          "100": "Bundesliga",
        },
      }}
    />
  );

  it("should render correctly", () => {
    const tree = renderer.create(predictionFilter).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
