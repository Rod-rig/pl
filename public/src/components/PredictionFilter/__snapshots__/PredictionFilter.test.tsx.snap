// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PredictionFilter should render correctly 1`] = `
<div>
  <div
    className="class_1-wrap-4"
  >
    <div
      className="MuiFormControl-root-5 class_1-control-3"
    >
      <label
        className="MuiFormLabel-root-20 MuiFormLabel-filled-24 MuiInputLabel-root-9 MuiInputLabel-formControl-14 MuiInputLabel-animated-17 MuiInputLabel-shrink-16"
        data-shrink={true}
        htmlFor="date"
      >
        Date
      </label>
      <div
        className="MuiInputBase-root-47 MuiInput-root-34 MuiInput-underline-38 MuiInputBase-formControl-48 MuiInput-formControl-35"
        onClick={[Function]}
      >
        <div
          className="MuiSelect-root-27"
        >
          <div
            aria-haspopup="true"
            aria-pressed="false"
            className="MuiSelect-select-28 MuiSelect-selectMenu-31 MuiInputBase-input-57 MuiInput-input-42"
            id="select-date"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            id="date"
            name="date"
            type="hidden"
            value="date"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-64 MuiSelect-icon-33"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root-103 MuiButton-root-77 MuiButton-contained-88 MuiButton-containedSecondary-90 MuiButton-raised-91 MuiButton-raisedSecondary-93 MuiButton-sizeSmall-100 class_1-btn-1"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-78"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-64 class_1-btnIcon-2"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
        Show
      </span>
      <span
        className="MuiTouchRipple-root-106"
      />
    </button>
  </div>
</div>
`;
